### PR TITLE
Player

### PR DESCRIPTION
**Pull request by t-h2o**
### New features

* Colission
* Ray front of the player

### CODE REVIEW

File: srcs/mini_map.c [hunk 1]
```diff
@@ -83,5 +83,6 @@ void	print_minimap(t_info *info)
 			print_square(info, info->map[pos_y][pos_x], pos_y, pos_x);
 	}
 	print_player(info);
+	player_ray(info);
```
Comment (Giggzzz) [inline srcs/mini_map.c:86]: The action to print the map starts the ray ?

File: srcs/player.c [hunk 2]
```diff
@@ -19,8 +19,14 @@
...
-	info->player.y += distance * info->player.dy;
+	float	next[2];
+
+	next[X] = info->player.x + distance * info->player.dx;
+	next[Y] = info->player.y + distance * info->player.dy;
+	if (info->map[(int)(next[Y])][(int)(next[X])] == '1')
+		return ;
+	info->player.x = next[X];
+	info->player.y = next[Y];
 	print_minimap(info);
```
Comment (Giggzzz) [inline srcs/player.c:30]: The player prints the map?

File: srcs/player.c [hunk 3]
```diff
@@ -34,7 +40,6 @@ void
 		info->player.angle += 2 * M_PI;
 	if (2 * M_PI < info->player.angle)
 		info->player.angle -= 2 * M_PI;
-	info->player.dx = cos(info->player.angle);
-	info->player.dy = sin(info->player.angle);
+	angle_delta(info->player.angle, &(info->player.dx), &(info->player.dy));
 	print_minimap(info);
```
Comment (Giggzzz) [inline srcs/player.c:44]: The player prints the map?

File: srcs/ray.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+// ray[X, Y]: The position of the ray
+static void	ray(t_info *info, float dx, float dy)
+{
+	float	ray[2];
+
+	ray[X] = info->player.x;
+	ray[Y] = info->player.y;
+	while (info->map[(int)(ray[Y])][(int)(ray[X])] != '1')
+	{
+		mlx_pixel_put(info->mlx[INIT], info->mlx[WINDOW],
```
Comment (Giggzzz) [inline srcs/ray.c:24]: could be in print minimap function

File: srcs/ray.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+			ray[X] * MM_SIZE_TILE + MM_POS_X,
+			ray[Y] * MM_SIZE_TILE + MM_POS_Y,
+			CO_BLUE);
+		ray[X] += dx / MM_SIZE_TILE;
+		ray[Y] += dy / MM_SIZE_TILE;
```
Comment (Giggzzz) [inline srcs/ray.c:29]: I'm not sure the precision (steps) of the ray should depends on the size of the minimap.

File: srcs/player.c (hunk 3 above)
Comment (t-h2o) [reply srcs/player.c:44]: Yes, when the player moves, it updates the player's position and then prints the mini map again.

File: srcs/ray.c (hunk 5 above)
Comment (t-h2o) [reply srcs/ray.c:29]: My bad, I used again the sprite size for my calcules, I will find another solution.

**Review by Giggzzz — APPROVED** [review]
We'll change some things later. Fine atm

**Pull request merged**